src/components/Dropdown/Dropdown.vue

```diff
@@ -236,7 +236,7 @@ export default {
       default: null
     }
   },
-  emits: ['update:modelValue', 'input', 'change'],
+  emits: ['update:modelValue', 'input', 'change', 'isDrawerOpen', 'hoverOverIndex'],
```

Comment:
I like the approach to this solution!  I would just suggest different names for these emits.  We should always try to name the emits as the events they are representing.  So maybe instead of `isDrawerOpen` we could add 2 emits for `open:drawer` and `close:drawer`. (or even `open:list` and `close:list`, I'm not sure `drawer` is typically used for Dropdowns as much as `list` but maybe?)  

And then `hoverOverIndex` could maybe be something like `hover:option`.